libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -1408,6 +1407,8 @@ protected virtual async Task<OAuthClient> CreateOAuthApiClientAsync(ITurnContext
                 OAuthClientConfig.EmulateOAuthCards = true;
             }
 
+            var oAuthEndpoint = OAuthClientConfig.OAuthEndpoint;
```

Comment:
The assumption is that we should not be calling these from different threads as we are still caching a static variable. A context switch before this line could result in incorrect endpoint being used if two threads set it to different values. I wonder if the approach is to do better documentation or improve synchronization here? Thoughts?

Reply:
Yes, there will always be a race when using a static like this.  As long as we're setting it globally via an FCB I don't think it will matter.  For this PR, I wanted to minimize changes and get something that will allow us to change it globally.  If we end up needing to set the value on a per-call basis then we'll need another PR to redesign the API and remove the static.